moda-contracts/src/CatalogFactory.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.21;
+
+import "@openzeppelin/contracts/proxy/beacon/BeaconProxy.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import {ICatalog} from "./interfaces/Catalog/ICatalog.sol";
+import {ICatalogFactory} from "./interfaces/Catalog/ICatalogFactory.sol";
+import {ICatalogInitialize} from "./interfaces/Catalog/ICatalogInitialize.sol";
+import {IModaRegistry} from "./interfaces/ModaRegistry/IModaRegistry.sol";
+import {IMembership} from "./interfaces/IMembership.sol";
+
+contract CatalogFactory is ICatalogFactory, AccessControl {
+    IModaRegistry public modaRegistry;
+    address public catalogBeacon;
+    string public version;
+
+    /**
+     * @dev Constructor
+     * @param modaRegistry_ - The address of the ModaRegistry contract
+     * @param catalogBeacon_ - The address of the catalog beacon that holds
+     * the Catalog implementation address
+     */
+    constructor(IModaRegistry modaRegistry_, address catalogBeacon_) {
+        _grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
+
+        modaRegistry = modaRegistry_;
+        catalogBeacon = catalogBeacon_;
+        version = "1";
+    }
+
+    /**
+     * @inheritdoc ICatalogFactory
+     */
+    function create(string calldata name, IMembership membership) external returns (address) {
+        address catalog = address(new BeaconProxy(catalogBeacon, ""));
+        ICatalogInitialize(catalog).initialize(
+            msg.sender, name, version, address(modaRegistry), membership
+        );
+        IModaRegistry(modaRegistry).registerCatalog(ICatalog(catalog));
+        emit CatalogCreated(msg.sender, catalog, name, version);
+
+        return catalog;
+    }
+
+    /**
+     * @inheritdoc ICatalogFactory
+     */
+    function setVersion(string calldata version_) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        version = version_;
```

Comment:
The version should be set in the Catalog itself. It is bug-prone to half to manage two different contracts and coordinate changes with the version number. It should be baked in to the Catalog so that as upgrades occur it either auto-increments, or we inject the version

Reply:
auto incrementing would be ideal, but not sure how feasible it would be without looking at the code